src/main/java/it/gov/pagopa/pu/registry/mapper/installment/DebtPositionIoEventDTO2InstallmentRegistryMapper.java

```diff
@@ -0,0 +1,44 @@
+package it.gov.pagopa.pu.registry.mapper.installment;
+
+import it.gov.pagopa.pu.registry.event.payments.dto.DebtPositionIoEventDTO;
+import it.gov.pagopa.pu.registry.model.InstallmentRegistry;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.stereotype.Service;
+
+import java.util.List;
+import java.util.stream.Collectors;
+
+@Slf4j
+@Service
+public class DebtPositionIoEventDTO2InstallmentRegistryMapper {
+
+  public List<InstallmentRegistry> map(DebtPositionIoEventDTO dto) {
+    if (dto.getPayload().getMessages() == null) return List.of();
+
+    List<String> iudList;
+
+    if (dto.getEventDescription().startsWith("IUD:")) {
+      iudList = List.of(dto.getEventDescription().split(":")[1].trim().split(","));
+    } else {
+      iudList = null;
+    }
+
+    return dto.getPayload().getMessages().stream()
+      .filter(ioMessageDTO -> {
+        if (iudList == null) return true;
+        return iudList.contains(ioMessageDTO.getNotificationId());
+      })
```

Comment:
You should consider only ioMessage having `nav` field not null or empty!
For this eventType, the IUD list is not necessary and not configured on the description field!
(documented in https://pagopa.atlassian.net/wiki/spaces/SPAC/pages/1601503397/Gestione+degli+Eventi)